src/utils/AuthContext.tsx

```diff
@@ -1,46 +1,87 @@
 import {
   AuthError,
-  AuthResponse,
   Session,
   User,
   UserAttributes,
-  UserResponse,
 } from '@supabase/supabase-js';
 import React, {
   createContext,
   useContext,
   useEffect,
   useMemo,
-  useState,
+  useReducer,
 } from 'react';
 
 import supabase from './supabase';
 
+type AuthContextAction =
+  | { type: 'LOADING' }
+  | { type: 'CLEAR_ERROR' }
+  | { type: 'SIGN_UP' }
+  | { type: 'SIGN_IN_WITH_EMAIL' }
+  | { type: 'VERIFY_OTP' }
+  | { type: 'RESEND_VERIFICATION' }
+  | { type: 'RESET_PASSWORD' }
+  | { type: 'REFRESH_SESSION'; session: Session | null }
+  | { type: 'UPDATE_USER'; user: User | null }
+  | { type: 'SIGN_OUT' };
+
+export type AuthDispatch = React.Dispatch<AuthContextAction>;
+
 export interface AuthState {
   session: Session | null;
   user: User | null;
```

Comment:
i think we can replace this user object stored in the context with the user profile. i feel like that might make more sense as it would likely be more commonly used in the app/other screens than the session.user object (which i'm seeing is only used in the home screen right now to get the id of the user, but the profile could accomplish this too). also there likely won't be frequent changes to this profile as i assume it'd only be changed during onboarding of the user and if they go to the settings to do something. also we can always just get the auth user object from session.user still. 

this will likely have implications with your UPDATE_USER action right now. from what i'm seeing this is used for resetting password, so maybe you can split it into another action and keep the UPDATE_USER one for updates to the profile object.